src/test/java/no/stelar7/api/r4j/tests/ratelimit/RatelimitTest.java

```diff
@@ -1,85 +1,119 @@
-            r4J.getDDragonAPI().getLanguages();
-            if (i % 9 == 0)
-            {
-                plat = LeagueShard.values()[plat.ordinal() + 1];
-            }
-            System.out.format("call no. %s Total time: %sms%n", i + 1, System.currentTimeMillis() - start);
-        }
-    }
-}
+package no.stelar7.api.r4j.tests.ratelimit;
+
+import no.stelar7.api.r4j.basic.calling.DataCall;
+import no.stelar7.api.r4j.basic.constants.api.regions.LeagueShard;
+import no.stelar7.api.r4j.impl.R4J;
+import no.stelar7.api.r4j.impl.lol.builders.spectator.SpectatorBuilder;
+import no.stelar7.api.r4j.impl.lol.builders.summoner.SummonerBuilder;
+import no.stelar7.api.r4j.pojo.lol.summoner.Summoner;
+import no.stelar7.api.r4j.tests.*;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+
+@ExtendWith(TimingExtension.class)
+public class RatelimitTest
+{
+    
+    final R4J r4J = new R4J(SecretFile.CREDS);
+    
+    @Test
+    @Disabled
+    public void testRateLimitThreaded()
+    {
+
+        DataCall.setCacheProvider(null);
+        try
+        {
+            String          name   = new SpectatorBuilder().withPlatform(LeagueShard.EUW1).getFeaturedGames().get(0).getParticipants().get(0).getSummonerName();
+            Summoner        s    = new SummonerBuilder().withPlatform(LeagueShard.EUW1).withName(name).get();
+            ExecutorService pool = Executors.newFixedThreadPool(8);
+            for (int i2 = 0; i2 < 130; i2++)
+            {
+                pool.execute(() -> new SummonerBuilder().withPlatform(LeagueShard.EUW1).withSummonerId(s.getSummonerId()).get());
+            }
+            pool.shutdown();
+            pool.awaitTermination(Long.MAX_VALUE, TimeUnit.DAYS);
+        } catch (InterruptedException e)
+        {
+            e.printStackTrace();
+        }
+    }
+    
+    @Test
+    @Disabled
+    public void testRateLimitVirtualThread()
+    {
+
+        DataCall.setCacheProvider(null);
+        try
+        {
+            String          name   = new SpectatorBuilder().withPlatform(LeagueShard.EUW1).getFeaturedGames().get(0).getParticipants().get(0).getSummonerName();
+            Summoner        s    = new SummonerBuilder().withPlatform(LeagueShard.EUW1).withName(name).get();
+            
+            List<Thread> threadToWait = new ArrayList<>();
+            
+            for (int i2 = 0; i2 < 130; i2++)
+            {
+                //Require Java 21
+                //threadToWait.add(Thread.ofVirtual().start(() -> new SummonerBuilder().withPlatform(LeagueShard.EUW1).withSummonerId(s.getSummonerId()).get()));
+            }
+            
+            for (Thread thread : threadToWait)
+            {
+                thread.join();
+            }
+        } catch (InterruptedException e)
+        {
+            e.printStackTrace();
+        }
+    }
+    

```

Comment:
This test doesnt do anything? (since nothing is added to `threadToWait`)

Reply:
Since R4J is in Java 8 and the tests require Java 21 to be "compiled" and run, I preferred to comment out the line that requires Java 21 to avoid showing irrelevant errors.

Uncomment the line to test this code, and it should be good. Would you have done it another way?